hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -3891,11 +3891,27 @@ public TableDescriptors getTableDescriptors() {
   /**
    * Reload the configuration from disk.
    */
-  void updateConfiguration() {
+  public void updateConfiguration() throws IOException {
     LOG.info("Reloading the configuration from disk.");
     // Reload the configuration from disk.
+    preUpdateConfiguration();
     conf.reloadConfiguration();
     configurationManager.notifyAllObservers(conf);
+    postUpdateConfiguration();
+  }
+
+  private void preUpdateConfiguration() throws IOException {
```

Comment:
Couldn't we make this and below method protected and override them in HMaster? The HMaster impl could call the master coprocessor hook

Reply:
+1 to this idea, this is how I did it on our private backport

Reply:
Oh. In that case, I'll bring over @charlesconnell 's patch from the private backport.